homeassistant/components/youless/sensor.py

```diff
@@ -0,0 +1,197 @@
+"""The sensor entity for the Youless integration."""
+from __future__ import annotations
+
+from youless_api.youless_sensor import YoulessSensor
+
+from homeassistant.components.youless import DOMAIN
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_DEVICE, DEVICE_CLASS_POWER
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import StateType
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Initialize the integration."""
+    coordinator = hass.data[DOMAIN][entry.entry_id]
+    device = entry.data[CONF_DEVICE]
+    if device is None:
+        device = entry.entry_id
+
+    async_add_entities(
+        [
+            GasSensor(coordinator, device),
+            PowerMeterSensor(coordinator, device, "low"),
+            PowerMeterSensor(coordinator, device, "high"),
+            PowerMeterSensor(coordinator, device, "total"),
+            CurrentPowerSensor(coordinator, device),
+            DeliveryMeterSensor(coordinator, device, "low"),
+            DeliveryMeterSensor(coordinator, device, "high"),
+            ExtraMeterSensor(coordinator, device, "total"),
+            ExtraMeterSensor(coordinator, device, "usage"),
+        ]
+    )
+
+
+class YoulessBaseSensor(CoordinatorEntity, Entity):
+    """The base sensor for Youless."""
+
+    def __init__(
+        self,
+        coordinator: DataUpdateCoordinator,
+        device: str,
+        device_group: str,
+        friendly_name: str,
+        sensor_id: str,
+    ) -> None:
+        """Create the sensor."""
+        super().__init__(coordinator)
+        self._device = device
+        self._device_group = device_group
+        self._sensor_id = sensor_id
+
+        self._attr_unique_id = f"{DOMAIN}_{device}_{sensor_id}"
```

Comment:
We don't need to include the integration domain in the unique id. The entity registry is aware of both the integration domain and the platform domain.